File: VB/WebSite/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Web.Mvc
Imports DXWebApplication1.Models
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace DevExpressMvcApplication1.Controllers
	Public Class HomeController
		Inherits Controller
		Private list As New PersonsList()
		Public Function Index() As ActionResult
			Return View(list.GetPersons())
		End Function

		Public Function GridViewPartial() As ActionResult
			Return PartialView(list.GetPersons())
		End Function

		<HttpPost, ValidateInput(False)> _
		Public Function EditingAddNew(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal personInfo As Person) As ActionResult
			If ModelState.IsValid Then
				list.AddPerson(personInfo)
			End If
			Return PartialView("GridViewPartial", list.GetPersons())
		End Function

		<HttpPost, ValidateInput(False)> _
		Public Function EditingUpdate(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal personInfo As Person) As ActionResult
			If ModelState.IsValid Then
				list.UpdatePerson(personInfo)
			End If
			Return PartialView("GridViewPartial", list.GetPersons())
		End Function

		Private Sub UpdateModelLastNameValue(ByVal personInfo As Person)
			If Request.Params("FirstName").Contains("text") Then
				personInfo.LastName = Request.Params("cmb")
			Else
				personInfo.LastName = Request.Params("txt")
			End If
			ModelState.Clear()
			UpdateModel(Of Person)(personInfo)
		End Sub

		Public Function EditingDelete(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal personId As Integer) As ActionResult
			list.DeletePerson(personId)
			Return PartialView("GridViewPartial", list.GetPersons())
		End Function

	End Class
End Namespace

File: CS/WebSite/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using DXWebApplication1.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace DevExpressMvcApplication1.Controllers {
    public class HomeController : Controller {
        PersonsList list = new PersonsList();
        public ActionResult Index() {
            return View(list.GetPersons());
        }

        public ActionResult GridViewPartial() {
            return PartialView(list.GetPersons());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult EditingAddNew([ModelBinder(typeof(DevExpressEditorsBinder))]Person personInfo) {
            if (ModelState.IsValid)
                list.AddPerson(personInfo);
            return PartialView("GridViewPartial", list.GetPersons());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult EditingUpdate([ModelBinder(typeof(DevExpressEditorsBinder))]Person personInfo) {
            if (ModelState.IsValid)
                list.UpdatePerson(personInfo);
            return PartialView("GridViewPartial", list.GetPersons());
        }

        private void UpdateModelLastNameValue(Person personInfo) {
            if (Request.Params["FirstName"].Contains("text"))
                personInfo.LastName = Request.Params["cmb"];
            else
                personInfo.LastName = Request.Params["txt"];
            ModelState.Clear();
            UpdateModel<Person>(personInfo);
        }

        public ActionResult EditingDelete([ModelBinder(typeof(DevExpressEditorsBinder))]int personId) {
            list.DeletePerson(personId);
            return PartialView("GridViewPartial", list.GetPersons());
        }

    }
}
